bin/trigger_azure_pipelines.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""A script which triggers and waits for Azure DevOps to complete a build"""
+
+import os
+import sys
+import time
+from ast import parse
+
+from azure.devops.connection import Connection
+from msrest.authentication import BasicAuthentication
+
+# https://learn.microsoft.com/en-us/rest/api/azure/devops/build/builds/get?view=azure-devops-rest-7.1#buildresult
+FAILED_STATUSES: set[str] = {"canceled", "partiallySucceeded", "failed"}
+COMPLETED_STATUSES: set[str] = FAILED_STATUSES | {"succeeded"}
+
+
+def create_ado_connection(organization, ado_pat) -> Connection:
+    """Returns an ADO connection to call the ADO REST APIs."""
+
+    return Connection(
+        base_url=f"https://dev.azure.com/{organization}",
+        creds=BasicAuthentication("", ado_pat),
+    )
+
+
+def build_status_match(build, expected_status_list):

```

Comment:
This doesn't seem to be used anywhere. Do we still need it?

Reply:
Thanks Melinda! Yup this is not needed.